docs/DeveloperGuide.md

```diff
@@ -99,28 +99,57 @@ Optionally, you can follow the [Using Checkstyle](UsingCheckstyle.md) document t
+### Javascript Files
+- **api.js** - loading and parsing of the dashboard content
+- **main.js** - main controller that handles the loading of the content into different modules
+- **v_summary.js** - module that supports the ramp chart view
+- **v_authorship.js** - module that supports the authorship view
 
-The dashboard contains two main parts: data `JSONs`(generated by `ReportGenerator`), and the static Dashboard template. Refer to [here](Build.md) for instructions on how to build the project and dashboard.
+### Overall Structure
+The main Vue object (`window.app`) is responsible for the loading of the dashboard (through `summary.json`). Its `repos` attribute is tied to the global `window.REPOS`, and is passed into the various other modules when the information is needed.
 
- * As shown in the graph above, there will be one summary `JSON`(`summary.json`) containing the summary information of all repositories, and one set of repo detail `JSON`(`commits.json` and `authorship.json`) for each repository, containing the contribution information of each specified author to that repository.
+`window.app` is broken down into two main parts, the summary view and the tabs view. The basic design is to have a summary view (with the ramp charts) as the main dashboard, and other modules display additional information in the tabbed interface which is displayed on the right.
 
-  * The dashboard template is a set of `HTML` + `CSS` + `Javascript`, which is located in `template/`. It is used to interpret the `JSON` information and display the dashboard on a web browser. <br>
-  Whenever *RepoSense* is built using *Gradle*, the current dashboard template's content will be zipped into `templateZip.zip` file for easier retrieval, and packaging into the generated `.jar` file.<br>
-  The dashboard template will then be unzipped to the target location along with the generated `JSON` data files in order to produce the dashboard whenever *RepoSense* generates a new report.
+**overall architecture**
+
+![overall architecture](images/dashboard-architecture.png)
+
+### Loading of dashboard information
+When the dashboard is first loaded, the main vue object tries to retreive the `summary.json` file in order to determine the right `commits.json` files to load into memory. `api.js` handles the loading of the file, and approriately gets the relevant file information, depending on whether the network files is available or a report archive have to be used.
+
+Once the relevant `commit.json` files hash been loaded, all the repo information will be passed into `v_summary` to be loaded in the summary view as the releveant ramp charts.
+
+### Activating additional view modules
+Most activity or actions should happen within the module itself, but in the case where there is a need to control the tab view from the module, an event is emitted from the module to main vue object (`window.app`), which then handles the data receive and pass it along to the relevant modules.
```

Comment:
from the module to **the** main **V**ue object ...
handles the data receive**d** and pass**es** it along ...